### PR TITLE
Improve workspace host error handling

### PR DESCRIPTION
**Pull request by nwalters512**
This PR makes a few improvements to workspace host error handling based on the errors we've seen in Sentry:

- Instead of writing the graded files archive to disk, we now directly pipe it to the response. This incidentally fixes an error where we were a) not awaiting the `fsPromises.unlink` return value and b) sometimes passing `undefined` to `fsPromises.unlink`
- The return value of `_getWorkspaceSettingsAsync` was previously unawaited; now it is.
- We weren't using the `new` keyword with `Stream.Transform`.

I also wired up Sentry request tracking for workspace host commands.

### CODE REVIEW

File: workspace_host/interface.js [hunk 1]
```diff
@@ -97,7 +98,7 @@ app.get(
     } else {
       res.status(200);
     }
-    res.jsonp({
+    res.json({
```
Comment (nwalters512) [inline workspace_host/interface.js:101]: AFAICT there's no legitimate reason to be using `jsonp` here, as this is strictly server-to-server communication.

File: workspace_host/interface.js [hunk 2]
```diff
@@ -392,7 +395,7 @@ async function pushContainerContentsToS3(workspace) {
 
   const workspacePath = path.join(workspacePrefix, `workspace-${workspace.launch_uuid}`);
   const s3Path = `workspace-${workspace.id}-${workspace.version}/current`;
-  const settings = _getWorkspaceSettingsAsync(workspace.id);
+  const settings = await _getWorkspaceSettingsAsync(workspace.id);
```
Comment (nwalters512) [inline workspace_host/interface.js:398]: I briefly enabled TypeScript error checking in this file, which is how I discovered this. There are too many other errors to leave it enabled permanently (mostly things with `util.callbackify`/`util.promisify`), but this was an easy one to fix.

File: workspace_host/interface.js [hunk 3]
```diff
@@ -1387,89 +1391,39 @@ function resetSequence(workspace_id, res) {
...
+      archive.file(filePath, { name: file });
+      debug(`Sending ${file}`);
+    } catch (err) {
+      logger.warn(`Graded file ${file} does not exist.`);
+      continue;
     }
-  );
+  }
+
+  await archive.finalize();
```
Comment (mwest1066) [inline workspace_host/interface.js:1428]: Wow, that's a lot less code! 🎉

**Review by mwest1066 — APPROVED** [review]
Thanks for starting to bring some sanity to the code in `interface.js`!

**Pull request merged**